src/maliput_malidrive/builder/direction_usage_builder.cc

```diff
@@ -68,13 +68,15 @@ DirectionUsageRule::State::Type DirectionUsageBuilder::ParseStateType(const std:
       {"Undefined", DirectionUsageRule::State::Type::kUndefined},
   };
 
-  MALIDRIVE_THROW_UNLESS(string_to_state.find(state) != string_to_state.end());
+  MALIDRIVE_VALIDATE(string_to_state.find(state) != string_to_state.end(), maliput::common::state_provider_error,
+                     "State type `" + state + "` does not exist.");
```

Comment:
this should be rulebook_error.

State provider error should be reserved to phase and rule state providers.

Reply:
Got it. Done.